samples/nrf9160/coap_client/README.rst

```diff
@@ -3,12 +3,24 @@
 nRF9160: nRF CoAP Client
 ########################
 
-The nRF CoAP Client sample demonstrates how to receive data from a public CoAP server with an nRF9160 SiP.
+The nRF CoAP Client sample demonstrates the communication between a public CoAP server and an nRF9160 SiP which acts as the CoAP client.
 
 Overview
-*********
+********
 
-The sample connects to a public CoAP test server, sends periodic GET request for a test resource that is available on the server, and prints the data that is received.
+The nRF CoAP Client sample performs the following actions:
+
+#. Connect to the configured public CoAP test server (indicated by the parameter ``CONFIG_COAP_SERVER_HOSTNAME``)
+#. Send periodic GET request for a test resource (indicated by the parameter ``CONFIG_COAP_RESOURCE``) that is available on the server
+#. Display the received data about the resource on a terminal emulator.
+
+The public CoAP server used in this sample is Californium CoAP server (coap://californium.eclipse.org:5683).
+This server runs Eclipse Californium which is an open source implementation of the CoAP protocol that is targeted at the development and testing of IoT applications.
+An nRF9160 DK board is used as the CoAP client.
+	
+An observable resource **obs** (Californium observable resource) is used in the communication between the CoAP client and the public CoAP server.
```

Comment:
Added some text about obs resource. 

Some questions to the developers: 

- Are we using only this resource in the sample? 
- Any other resources can be configured?
- If yes, how do we do it? Is there any limit on the number of observable resources that can be configured?

Reply:
> Are we using only this resource in the sample?

Yes

> Any other resources can be configured?
> If yes, how do we do it? Is there any limit on the number of observable resources that can be configured?

The sample will query one resource at a time. It can be changed with `CONFIG_COAP_RESOURCE` Kconfig option. And the resource does not have to be observable. The sample uses the standard request/response mechanism, not the observe mechanism.